src/sortStudents.ts

```diff
@@ -1,16 +1,61 @@
 
 export interface Student {
   // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
   // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc'| 'desc';
 
-
-export function sortStudents(students, sortBy, order) {
+export function sortStudents(
+  students : Student[], sortBy: SortType, order: SortOrder,
+) : Student[] {
   // write your function
+  let resultArray : Student[] = [...students];
+
+  function callbackGrades(
+    previousValue: number, currentValue: number,
+  ) : number {
+    return previousValue + currentValue;
+  }
+
+  switch (sortBy) {
+    case 'age':
+      resultArray = resultArray.sort((a, b) => +a.age - +b.age);
```

Comment:
`.sort()` is not immutable
```suggestion
      resultArray.sort((a, b) => +a.age - +b.age);
```